ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ComplianceLevelsByStandard.tsx

```diff
@@ -0,0 +1,188 @@
+} from 'constants/entityTypes';
+import { AGGREGATED_RESULTS_ACROSS_ENTITIES } from 'queries/controls';
+import { complianceBasePath, urlEntityListTypes } from 'routePaths';
+import { standardLabels } from 'messages/standards';
+import ComplianceLevelsByStandardChart, { ComplianceData } from './ComplianceLevelsByStandardChart';
+import WidgetCard from './WidgetCard';
+
+const fieldIdPrefix = 'compliance-levels-by-standard';
+
+function getViewAllLink(searchFilter: SearchFilter) {
+    return `${complianceBasePath}?${getUrlQueryStringForSearchFilter(searchFilter)}`;
+}
+
+// Adapted from `processData` function in the original DashboardCompliance.js code
+function processData(
+    searchFilter: SearchFilter,
+    sortDirection: SortBy,
+    data?: AggregationResult
+): ComplianceData | undefined {
+    if (!data) {
+        return undefined;
+    }
+    const { complianceStandards } = data;
+    const modifiedData = data.controls.results.map((result) => {
+        const aggregationName = result?.aggregationKeys[0]?.id;
+        const standard = complianceStandards.find((cs) => cs.id === aggregationName);
+        const { numPassing, numFailing } = result;
+        const standardQueryValue =
+            standardLabels[standard?.id] || aggregationName || 'Unrecognized standard';
+        const query = getUrlQueryStringForSearchFilter({
+            ...searchFilter,
+            Cluster: searchFilter.Cluster || '*',
+            standard: standardQueryValue,
+        });
+        const link = `${complianceBasePath}/${
+            urlEntityListTypes[standardEntityTypes.CONTROL]
+        }?${query}`;
+        const modifiedResult = {
+            name: standard?.name || aggregationName || 'Unrecognized standard',
+            passing: Math.round((numPassing / (numFailing + numPassing)) * 100) || 0,
+            link,
+        };
+        return modifiedResult;
+    });
+
+    const sorted = sortBy(modifiedData, [(datum) => datum.passing]);
+
+    if (sortDirection === 'asc') {
+        sorted.reverse();
+    }
+
+    return sorted;
+}
+
+type AggregationResult = {
+    controls: {
+        results: {
+            aggregationKeys: {
+                id: keyof typeof standardTypes;
+                scope: StandardEntityType;
+            }[];
+            numFailing: number;
+            numPassing: number;
+            numSkipped: number;
+            unit: StandardEntityType;
+        }[];
+    };
+    complianceStandards: {
+        id: string;
+        name: string;
+    }[];
+};
+
+type SortBy = 'asc' | 'desc';
+
+function ComplianceLevelsByStandard() {
+    const { isOpen: isOptionsOpen, onToggle: toggleOptionsOpen } = useSelectToggle();
+    const { searchFilter } = useURLSearch();
+    const [sortDataBy, setSortDataBy] = useState<SortBy>('asc');
+
+    const where = getRequestQueryStringForSearchFilter({
+        // We always need to include some value for Cluster, otherwise aggregation will be performed at the namespace level
```

Comment:
The aggregation done on the back end will be different depending on the search parameters passed with the query. For example, if there is no value for `Cluster`, aggregation will be done at the namespace level and will be missing some controls. If we always pass a value for `Cluster`, such as `*` when all clusters are selected we can ensure the resulting data is consistent.

The returned data will still be filtered correctly by namespaces if those are provided in the search filter, but will be aggregated at the "cluster" level.